src/app/app.component.html

```diff
@@ -252,14 +261,27 @@
                 />
                 <line *ngFor="let x of xGrid; trackBy:trackByIndex"
                     class="grid"
+                    [ngClass]="{'tick': showTicks && x % tickInterval === 0 }"
                     [attr.x1]="x"
                     [attr.y1]="viewPortY"
                     [attr.x2]="x"
                     [attr.y2]="viewPortY + viewPortHeight"
                     [attr.stroke-width]="strokeWidth"
                 />
+
+                <ng-container *ngIf="!preview && showTicks">
+                    <ng-container *ngFor="let y of yGrid; trackBy:trackByIndex">
+                        <text class="tick" *ngIf="y % tickInterval ==  0" x="-0.15" [attr.y]="y - 0.15">{{ y }}</text>
+                    </ng-container>
+
+                    <ng-container *ngFor="let x of xGrid; trackBy:trackByIndex">
+                        <text class="tick" *ngIf="x != 0 && x % tickInterval == 0" y="-0.15" [attr.x]="x - 0.15">{{ x }}</text>
```

Comment:
For the tick labels, I would rather use a dynamic font-size, so that it works better with zoom.
Something like:
`[attr.x]="-4*strokeWidth" [attr.y]="y - 4*strokeWidth" [ngStyle]="{'font-size': strokeWidth*14+'px'}"`

(and then remove `font-size: 0.7px;` in CSS)

Reply:
This works great, thanks for the ideal revision